In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [5]:
df=pd.read_excel("Iris_Dataset.xlsx")
df.head()

,Sepal_Length_cm,Sepal_Width_cm,Petal_Length_cm,Petal_Width_cm,Species
0,6.3,3.3,4.7,1.6,versicolor
1,5.1,3.5,1.4,0.2,setosa
2,4.5,2.3,1.3,0.3,setosa
3,6.6,2.9,4.6,1.3,versicolor
4,4.9,3.1,1.5,0.2,setosa


In [6]:
df.shape

(80, 5)

In [7]:
df.columns

Index(['Sepal_Length_cm', 'Sepal_Width_cm', 'Petal_Length_cm',
       'Petal_Width_cm', 'Species'],
      dtype='str')

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 80 entries, 0 to 79
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Sepal_Length_cm  80 non-null     float64
 1   Sepal_Width_cm   80 non-null     float64
 2   Petal_Length_cm  80 non-null     float64
 3   Petal_Width_cm   80 non-null     float64
 4   Species          80 non-null     str    
dtypes: float64(4), str(1)
memory usage: 3.9 KB


In [9]:
df.isnull().sum()

Sepal_Length_cm    0
Sepal_Width_cm     0
Petal_Length_cm    0
Petal_Width_cm     0
Species            0
dtype: int64

In [11]:
numeric_cols=[
    "Sepal_Length_cm",
    "Sepal_Width_cm",
    "Petal_Length_cm",
    "Petal_Width_cm"
]
numeric_df=df[numeric_cols]

In [12]:
numeric_df.describe()

,Sepal_Length_cm,Sepal_Width_cm,Petal_Length_cm,Petal_Width_cm
count,80.000000,80.000000,80.000000,80.000000
mean,5.880000,3.078750,3.775000,1.196250
std,0.908386,0.368608,1.792495,0.756481
min,4.300000,2.300000,1.100000,0.100000
25%,5.075000,2.900000,1.500000,0.300000
50%,5.850000,3.000000,4.350000,1.300000
75%,6.625000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


In [13]:
numeric_df.mean()

Sepal_Length_cm    5.88000
Sepal_Width_cm     3.07875
Petal_Length_cm    3.77500
Petal_Width_cm     1.19625
dtype: float64

In [14]:
numeric_df.median()

Sepal_Length_cm    5.85
Sepal_Width_cm     3.00
Petal_Length_cm    4.35
Petal_Width_cm     1.30
dtype: float64

In [15]:
numeric_df.mode().iloc[0]

Sepal_Length_cm    5.0
Sepal_Width_cm     3.0
Petal_Length_cm    1.4
Petal_Width_cm     0.2
Name: 0, dtype: float64

In [16]:
numeric_df.std()

Sepal_Length_cm    0.908386
Sepal_Width_cm     0.368608
Petal_Length_cm    1.792495
Petal_Width_cm     0.756481
dtype: float64

In [17]:
numeric_df.quantile([0.25,0.50,0.75])

,Sepal_Length_cm,Sepal_Width_cm,Petal_Length_cm,Petal_Width_cm
0.25,5.075,2.9,1.50,0.3
0.50,5.850,3.0,4.35,1.3
0.75,6.625,3.3,5.10,1.8


In [18]:
IQR=numeric_df.quantile(0.75)-numeric_df.quantile(0.25)
IQR

Sepal_Length_cm    1.55
Sepal_Width_cm     0.40
Petal_Length_cm    3.60
Petal_Width_cm     1.50
dtype: float64

OVERALL STATISTICAL SUMMARY

In [20]:
summary=pd.DataFrame({
    "MEAN":numeric_df.mean(),
    "MEDIAN":numeric_df.median(),
    "MODE":numeric_df.mode().iloc[0],
    "STANDARD DEVIATION":numeric_df.std(),
    "MINIMUM":numeric_df.min(),
    "MAXIMUM":numeric_df.max(),
    "25th Percentile":numeric_df.quantile(0.25),
    "50th Percentile":numeric_df.quantile(0.50),
    "75th Percentile":numeric_df.quantile(0.75),
    "IQR":numeric_df.quantile(0.75)-numeric_df.quantile(0.25)
})
summary

,MEAN,MEDIAN,MODE,STANDARD DEVIATION,MINIMUM,MAXIMUM,25th Percentile,50th Percentile,75th Percentile,IQR
Sepal_Length_cm,5.88000,5.85,5.0,0.908386,4.3,7.9,5.075,5.85,6.625,1.55
Sepal_Width_cm,3.07875,3.00,3.0,0.368608,2.3,4.4,2.900,3.00,3.300,0.40
Petal_Length_cm,3.77500,4.35,1.4,1.792495,1.1,6.9,1.500,4.35,5.100,3.60
Petal_Width_cm,1.19625,1.30,0.2,0.756481,0.1,2.5,0.300,1.30,1.800,1.50


Sepal Length

Mean   = 5.88
Median = 5.85
SD     = 0.9083
IQR    = 1.55

Sepal Length has a mean of 5.88 cm and a median of 5.85 cm. Since the mean and median are very close, the distribution does not appear to have strong skewness based on these measures alone. The standard deviation of approximately 0.9 cm indicates moderate variation, while the IQR of 1.55 cm represents the spread of the middle 50% of observations.

Sepal Width

Mean   = 3.0788
Median = 3.00
SD     = 0.3686
IQR    = 0.40


Sepal Width has a mean of approximately 3.08 cm and a median of 3.00 cm. The relatively small difference between the mean and median suggests limited skewness. Its standard deviation of approximately 0.36 cm indicates relatively low variation compared with the other numerical features. The middle 50% of observations fall between 2.90 cm and 3.30 cm.

Petal Length

Mean   = 3.775
Median = 4.35
SD     = 1.7924
IQR    = 3.60

Petal Length has a mean of 3.775 cm and a median of 4.35 cm. The difference between the mean and median indicates that the distribution is not symmetric and may show skewness. Its standard deviation of approximately 1.8 cm and IQR of 3.60 cm indicate substantial variability in the measurements.

Petal Width

Mean   = 1.1962
Median = 1.30
SD     = 0.7564
IQR    = 1.50

Petal Width has a mean of approximately 1.20 cm and a median of 1.30 cm. The difference between these values indicates some asymmetry in the distribution. The standard deviation of approximately 0.75 cm indicates variation in petal width, while the IQR of 1.50 cm represents the spread of the middle 50% of observations.

Conclusion

Descriptive statistics provide a simple way to summarize and understand numerical data. In this analysis, mean, median, mode, standard deviation, percentiles, and IQR were calculated for the four numerical features of the Iris dataset. Comparing mean and median helped assess distribution characteristics, while standard deviation and IQR helped understand variability and spread. 